### PR TITLE
refactor(FRE-880): increase API chains data usage

### PR DESCRIPTION
**Pull request by codingki**
Fee assets, explorerUrl, pretty name now comes from API

### CODE REVIEW

File: packages/widget/src/hooks/use-swap-widget.ts [hunk 1]
```diff
@@ -620,11 +618,9 @@ export function useSwapWidget(persistSwapWidgetState = true) {
...
           });
 
-          feeDenomPrices ??= (await getChainFeeAssets(srcChain.chainID)).find(
-            ({ denom }) => {
-              return denom === srcFeeAsset?.denom;
-            }
-          );
+          feeDenomPrices ??= srcChain.feeAssets.find(({ denom }) => {
+            return denom === srcFeeAsset?.denom;
+          });
```
Comment (toddkao) [inline packages/widget/src/hooks/use-swap-widget.ts:623]: this looks redudant with lines 617-619

**Review by toddkao — APPROVED** [review]
I'm finding alot of the logic in `use-swap-widget` surprisingly difficult to follow, but as far as the refactor is concerned this looks good. The logic is definitely something I'll want to see if we can simplify in v2

File: packages/widget/src/hooks/use-swap-widget.ts (hunk 1 above)
Comment (codingki) [reply packages/widget/src/hooks/use-swap-widget.ts:623]: right, nice catch

**Comment by codingki** [comment]
> I'm finding alot of the logic in `use-swap-widget` surprisingly difficult to follow, but as far as the refactor is concerned this looks good. The logic is definitely something I'll want to see if we can simplify in v2

I agree with you, it is difficult to follow. I'm avoiding to refactor this legacy code. Yes we need to simplify it in v2

**Pull request merged**